### PR TITLE
Ip transfer

### LINKED ISSUE #260: Implement Linode networking swap ip tab

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #260, depends on #1357

screen shot 2017-03-08 at 7 06 25 pm

### CODE REVIEW

File: src/api/linodes.js [hunk 1]
```diff
@@ -180,3 +181,18 @@ export function linodeBackups(linodeId) {
...
   };
 }
+
+export function setShared(linodeId, ips) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const { token } = state.authentication;
+    await fetch(token, `/linode/instances/${linodeId}/ips/sharing`, {
+      method: 'POST',
+      body: JSON.stringify({ ips }),
```
Comment (mparke) [inline src/api/linodes.js]: move JSON.stringify for `body` into fetch?

File: src/api/apiActionReducerGenerator.js [hunk 2]
```diff
@@ -276,3 +276,23 @@ export default function apiActionReducerGenerator(config, actions) {
   thunks.type = config.plural;
   return thunks;
 }
+
+// Helpers function when making calls outside of the ability of the above thunks.
+function _thunkFetch(method) {
```
Comment (mparke) [inline src/api/apiActionReducerGenerator.js]: link follow-up issue?

Comment (eatonphil) [inline src/api/apiActionReducerGenerator.js]: https://github.com/linode/manager/issues/1370

File: src/api/linodes.js (hunk 1 above)
Comment (eatonphil) [inline src/api/linodes.js]: This is a remnant from pre-rebasing after IP Sharing was merged. Once I rebase this diff will go away.

**Pull request merged**